subscription-service/src/main/kotlin/com/egm/stellio/subscription/web/SubscriptionHandler.kt

```diff
@@ -92,33 +95,44 @@ class SubscriptionHandler(
         )
 
         return if (prevAndNextLinks.first != null && prevAndNextLinks.second != null)
-            ResponseEntity.status(HttpStatus.OK).header("Link", prevAndNextLinks.first)
-                .header("Link", prevAndNextLinks.second).body(subscriptions)
+            buildGetSuccessResponse(mediaType, contextLink)
+                .header(HttpHeaders.LINK, prevAndNextLinks.first)
+                .header(HttpHeaders.LINK, prevAndNextLinks.second)
+                .body(subscriptions)
         else if (prevAndNextLinks.first != null)
-            ResponseEntity.status(HttpStatus.OK).header("Link", prevAndNextLinks.first).body(subscriptions)
+            buildGetSuccessResponse(mediaType, contextLink)
+                .header(HttpHeaders.LINK, prevAndNextLinks.first)
+                .body(subscriptions)
         else if (prevAndNextLinks.second != null)
-            ResponseEntity.status(HttpStatus.OK).header("Link", prevAndNextLinks.second).body(subscriptions)
+            buildGetSuccessResponse(mediaType, contextLink)
+                .header(HttpHeaders.LINK, prevAndNextLinks.second)
+                .body(subscriptions)
         else
-            ResponseEntity.status(HttpStatus.OK).body(subscriptions)
+            buildGetSuccessResponse(mediaType, contextLink)
+                .body(subscriptions)
     }
 
     /**
      * Implements 6.11.3.1 - Retrieve Subscription
      */
     @GetMapping("/{subscriptionId}", produces = [MediaType.APPLICATION_JSON_VALUE, JSON_LD_CONTENT_TYPE])
     suspend fun getByURI(
+        @RequestHeader httpHeaders: HttpHeaders,
         @PathVariable subscriptionId: String,
         @RequestParam options: Optional<String>
     ): ResponseEntity<*> {
         val includeSysAttrs = options.filter { it.contains("sysAttrs") }.isPresent
+        val contextLink = getContextFromLinkHeaderOrDefault(httpHeaders)
+        val mediaType = getApplicableMediaType(httpHeaders)
         val subscriptionIdUri = subscriptionId.toUri()
         checkSubscriptionExists(subscriptionIdUri).awaitFirst()
 
         val userId = extractSubjectOrEmpty().awaitFirst()
         checkIsAllowed(subscriptionIdUri, userId).awaitFirst()
         val subscription = subscriptionService.getById(subscriptionIdUri).awaitFirst()
 
-        return ResponseEntity.status(HttpStatus.OK).body(subscription.toJson(includeSysAttrs))
+        return buildGetSuccessResponse(mediaType, contextLink)
+            .body(subscription.toJson(includeSysAttrs))
```

Comment:
Same here, the mediaType is not used for the response payload

Reply:
Also indeed.

Reply:
Done